video/out/x11_common.c

```diff
@@ -1606,10 +1609,16 @@ static void vo_x11_map_window(struct vo *vo, struct mp_rect rc)
     }
 
     if (x11->opts->all_workspaces || x11->opts->geometry.ws > 0) {
-        long v = x11->opts->all_workspaces
-                ? 0xFFFFFFFF : x11->opts->geometry.ws - 1;
-        XChangeProperty(x11->display, x11->window, XA(x11, _NET_WM_DESKTOP),
-                        XA_CARDINAL, 32, PropModeReplace, (unsigned char *)&v, 1);
+        if (x11->wm_type & vo_wm_STICKY) {
```

Comment:
this looks incorrect, `x11->opts->geometry.ws > 0` should still use _NET_WM_DESKTOP with the specific workspace

Reply:
i couldn't  find what `x11->opts->geometry.ws` is, so i left it as it is.
what about
```patch
---
 video/out/x11_common.c | 21 ++++++++++-----------
 1 file changed, 10 insertions(+), 11 deletions(-)

diff --git a/video/out/x11_common.c b/video/out/x11_common.c
index 37c9ba1ca37..98b4635a001 100644
--- a/video/out/x11_common.c
+++ b/video/out/x11_common.c
@@ -1608,17 +1608,16 @@ static void vo_x11_map_window(struct vo *vo, struct mp_rect rc)
         x11_send_ewmh_msg(x11, "_NET_WM_FULLSCREEN_MONITORS", params);
     }
 
-    if (x11->opts->all_workspaces || x11->opts->geometry.ws > 0) {
-        if (x11->wm_type & vo_wm_STICKY) {
-            Atom state = XA(x11, _NET_WM_STATE_STICKY);
-            XChangeProperty(x11->display, x11->window, XA(x11, _NET_WM_STATE), XA_ATOM,
-                            32, PropModeAppend, (unsigned char *)&state, 1);
-        } else {
-            long v = x11->opts->all_workspaces
-                    ? 0xFFFFFFFF : x11->opts->geometry.ws - 1;
-            XChangeProperty(x11->display, x11->window, XA(x11, _NET_WM_DESKTOP),
-                            XA_CARDINAL, 32, PropModeReplace, (unsigned char *)&v, 1);
-        }
+    if (x11->opts->geometry.ws > 0) {
+        long v = x11->opts->geometry.ws - 1;
+        XChangeProperty(x11->display, x11->window, XA(x11, _NET_WM_DESKTOP),
+                        XA_CARDINAL, 32, PropModeReplace, (unsigned char *)&v, 1);
+    }
+
+    if (x11->opts->all_workspaces && (x11->wm_type & vo_wm_STICKY)) {
+        Atom state = XA(x11, _NET_WM_STATE_STICKY);
+        XChangeProperty(x11->display, x11->window, XA(x11, _NET_WM_STATE), XA_ATOM,
+                        32, PropModeReplace, (unsigned char *)&state, 1);
     }
 
     vo_x11_update_composition_hint(vo);

```

Reply:
i sorry, but why?

Reply:
the current behaviour is that setting a workspace (`ws`) takes precedence over the all-workspaces option
this should be preserved

Reply:
i think the current behavior give precedence to all-workspaces option 
```c
    if (x11->opts->all_workspaces || x11->opts->geometry.ws > 0) {
        long v = x11->opts->all_workspaces
                ? 0xFFFFFFFF : x11->opts->geometry.ws - 1;
```

Reply:
i think the current behavior give precedence to all-workspaces option 
```c
    if (x11->opts->all_workspaces || x11->opts->geometry.ws > 0) {
        long v = x11->opts->all_workspaces
                ? 0xFFFFFFFF : x11->opts->geometry.ws - 1;
```

Reply:
oh yeah you're correct, then order needs to be swapped